lambda/translcore.ml

```diff
@@ -530,6 +569,7 @@ and transl_exp0 ~in_new_scope ~scopes e =
         ap_func=
           Lprim(Pfield 0, [transl_class_path loc e.exp_env cl], loc);
         ap_args=[lambda_unit];
+        ap_position=Apply_tail;
```

Comment:
It's always safe to treat it as `Apply_tail` because the function and argument are `Alloc_heap`. Doing it unconditionally saves us from needing to track which `new`s are in tail position.